File: DekafTestDB/src/core/MysqlTestHelper.java
package org.jetbrains.dekaf.core;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.dekaf.text.Scriptum;



/**
 * @author Leonid Bushuev from JetBrains
 **/
public class MysqlTestHelper extends BaseTestHelper<DBFacade> {

  public MysqlTestHelper(@NotNull final DBFacade db) {
    super(db, Scriptum.of(MysqlTestHelper.class));
    getSchemasNotToZap().add("performance_schema");
    getSchemasNotToZap().add("mysql");
  }


  @Override
  public void prepareX1() {
    performCommand("create or replace view X1 as select 1");
  }

  @Override
  public void prepareX1000() {
    performCommand(getScriptum(), "X10");
    performCommand(getScriptum(), "X1000");
  }

  @Override
  public void prepareX1000000() {
    prepareX1000();
    performCommand(getScriptum(), "X1000000");
  }


  @Override
  protected void ensureNoTableOrView4(final String[] names) {
    // Unfortunately, MS SQL provides no way to easy drop tables.
    // We have to drop foreign keys first.
    performMetaQueryCommands(getScriptum(), "EnsureNoForeignKeysMetaQuery", names);
    performMetaQueryCommands(getScriptum(), "EnsureNoTableOrViewMetaQuery", names);
  }

  @Override
  protected void zapSchemaInternally(final ConnectionInfo connectionInfo) {
    // Unfortunately, MS SQL provides no way to easy drop tables.
    // We have to drop foreign keys first.
    performMetaQueryCommands(getScriptum(), "ZapForeignKeysMetaQuery");
    performMetaQueryCommands(getScriptum(), "ZapSchemaMetaQuery");
  }

}

File: dev/dekaf-test-db/src/org/jetbrains/dekaf/core/MysqlTestHelper.kt
package org.jetbrains.dekaf.core



/**
 * MySQL test helper.
 * 
 * @author Leonid Bushuev
 */
class MysqlTestHelper(db: DBFacade) : BaseTestHelper(db, MysqlTestHelper::class) {

    init {
        schemasNotToZap.add("performance_schema")
        schemasNotToZap.add("mysql")
    }


    override fun prepareX1() {
        performCommand("create or replace view X1 as select 1")
    }

    override fun prepareX1000() {
        performCommand(scriptum, "X10")
        performCommand(scriptum, "X1000")
    }

    override fun prepareX1000000() {
        prepareX1000()
        performCommand(scriptum, "X1000000")
    }


    override fun ensureNoTableOrView(vararg names: String) {
        // Unfortunately, MySQL provides no way to easy drop tables.
        // We have to drop foreign keys first.
        performMetaQueryCommands(scriptum, "EnsureNoForeignKeysMetaQuery", *names)
        performMetaQueryCommands(scriptum, "EnsureNoTableOrViewMetaQuery", *names)
    }

    override fun zapSchemaInternally(connectionInfo: ConnectionInfo) {
        // Unfortunately, MySQL provides no way to easy drop tables.
        // We have to drop foreign keys first.
        performMetaQueryCommands(scriptum, "ZapForeignKeysMetaQuery")
        performMetaQueryCommands(scriptum, "ZapSchemaMetaQuery")
    }

}
